### PR TITLE
chore(SimpleList):convert to typescript

### LINKED ISSUE #8154: Simple list

### PR DESCRIPTION
**Pull request by janwright73**
**What**: Closes #8154

**Additional issues**:

### CODE REVIEW

**Review by gitdallas — APPROVED** [review]

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { SimpleList, SimpleListItem } from '@patternfly/react-core';
+
+export const SimpleListUncontrolledDemo: React.FunctionComponent = () => {
+  const [activeItem, setActiveItem] = React.useState(0);
+
+  const onSelect = (selectedItem: any, selectedItemProps: any) => {
```
Comment (nicolethoen) [inline packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx:7]: can we update the props' types in this `onSelect` handler to match the `SimpleList` prop types? 
```
onSelect?: (
    ref: React.RefObject<HTMLButtonElement> | React.RefObject<HTMLAnchorElement>,
    props: SimpleListItemProps
  ) => void;
```

**Review by nicolethoen — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx (hunk 1 above)
Comment (janwright73) [reply packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx]: agree, did more work to match this request

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { SimpleList, SimpleListItem } from '@patternfly/react-core';
+
+export const SimpleListUncontrolledDemo: React.FunctionComponent = () => {
+  const [activeItem, setActiveItem] = React.useState(0);
+
+  const onSelect = (
+    selectedItem: React.RefObject<HTMLButtonElement> | React.RefObject<HTMLAnchorElement>,
+    _selectedItemProps
```
Comment (nicolethoen) [inline packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx:9]: I think the type of the selectedItemProps is `SimpleListItemProps`. 

We usually use the underscore at the beginning of parameters as a cue that the parameter is not going to be used in the function body. In this case, the selectedItem is actually not used in the body, but the selectedItemProps param is, so i'd remove the underscore from the param name.

File: packages/react-core/src/components/SimpleList/examples/SimpleListBasic.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { SimpleList, SimpleListItem } from '@patternfly/react-core';
+
+export const SimpleListDemo: React.FunctionComponent = () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/SimpleList/examples/SimpleListBasic.tsx:4]: Just to match the example/file name:

```suggestion
export const SimpleListBasic: React.FunctionComponent = () => {
```

File: packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { SimpleList, SimpleListItem, SimpleListGroup } from '@patternfly/react-core';
+
+export const SimpleListGroupDemo: React.FunctionComponent = () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx:4]:
```suggestion
export const SimpleListGrouped: React.FunctionComponent = () => {
```

File: packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      List item 2
+    </SimpleListItem>,
+    <SimpleListItem key="item7" component="a" href="#">
+      List item 3
+    </SimpleListItem>,
+    <SimpleListItem key="item8">List item 4</SimpleListItem>
+  ];
+
+  return (
+    <SimpleList aria-label="Grouped List Example">
```
Comment (thatblindgeye) [inline packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx:26]:
```suggestion
    <SimpleList aria-label="Grouped simple list example">
```
To follow how the other aria-labels are formatted + the component name added in

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { SimpleList, SimpleListItem } from '@patternfly/react-core';
+
+export const SimpleListUncontrolledDemo: React.FunctionComponent = () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx:4]:
```suggestion
export const SimpleListUncontrolled: React.FunctionComponent = () => {
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Only had a few small change requests below

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx (hunk 6 above)
Comment (janwright73) [reply packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx:4]: name change accepted

File: packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx (hunk 5 above)
Comment (janwright73) [reply packages/react-core/src/components/SimpleList/examples/SimpleListGrouped.tsx:26]: text change accepted

File: packages/react-core/src/components/SimpleList/examples/SimpleListBasic.tsx (hunk 3 above)
Comment (janwright73) [reply packages/react-core/src/components/SimpleList/examples/SimpleListBasic.tsx:4]: name change accepted

File: packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx (hunk 2 above)
Comment (janwright73) [reply packages/react-core/src/components/SimpleList/examples/SimpleListUncontrolled.tsx:9]: got it - thanks

**Review by thatblindgeye — APPROVED** [review]
Looks good! Regarding the arialabel comment I had left, I opened #8249 as a followup.

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**